model-optimizer/mo/utils/cli_parser.py

```diff
@@ -1127,6 +1130,85 @@ def get_absolute_path(path_to_file: str) -> str:
+        return True
+    except ValueError:
+        return False
+
+
+def detect_value_type(value: str):
+    values = value.split(',')
+    for i in range(len(values)):
+        value = values[i]
+        if value.isdigit():
+            values[i] = int(value)
+        elif isfloat(value):
+            values[i] = float(value)
+
+    return values[0] if len(values) == 1 else values
+
+
+def parse_transform(transform: str, ie_is_available=True) -> list:
+    transforms = []
+
+    if len(transform) == 0:
+        return transforms
+
+    all_transforms = re.findall(r"([a-zA-Z0-9]+)(\[([^\]]+)\])*(,|$)", transform)
+
+    # Check that all characters were matched otherwise transform key value is invalid
+    key_len = len(transform)
+    for transform in all_transforms:
+        # In regexp we have 4 groups where 1st group - transformation_name,
+        #                                  2nd group - [args],
+        #                                  3rd group - args, <-- nested group
+        #                                  4th group - EOL
+        # And to check that regexp matched all string we decrease total length by the length of matched groups (1,2,4)
+        # In case if no arguments were given to transformation then 2nd and 3rd groups will be empty.
+        if len(transform) != 4:
+            raise Error("Unexpected transform key structure: {}".format(transform))
+        key_len -= len(transform[0]) + len(transform[1]) + len(transform[3])
+
+    if key_len != 0:
+        raise Error("Unexpected transform key structure: {}".format(transform))
+
+    for transform in all_transforms:
+        name = transform[0]
+        args = transform[2]
+
+        args_dict = {}
+
+        if len(args) != 0:
+            for arg in args.split(';'):
+                m = re.match(r"^([_a-zA-Z]+)=(.+)$", arg)
+                if not m:
+                    raise Error("Unrecognized attributes for transform key: {}".format(transform))
+
+                args_dict[m.group(1)] = detect_value_type(m.group(2))
+
+        transforms.append((name, args_dict))
+
+    if not ie_is_available and len(transforms) != 0:
+        raise Error('Can not apply {} transformations due to missing Inference Engine Python API'.format(
+            ','.join([name for name, _ in transforms])))
+
+    from mo.back.offline_transformations import get_available_transformations
+    available_transforms = get_available_transformations()
+
+    missing_transformations = []
+    for name, _ in transforms:
+        if name not in available_transforms.keys():
+            missing_transformations.append(name)
+
+    if len(missing_transformations) != 0:
+        raise Error('Following transformations ({}) are not available. List with available transformations ({})'.format(','.join(missing_transformations),
+                                                                                                                        ','.join(available_transforms.keys())))
```

Comment:
This should not be a part of this function. Please, create a separate function to get available transforms and validate transformations provided by the user.

Reply:
agree. changed.